### PR TITLE
closes #109

### LINKED ISSUE #109: Add test case for OPTIONS API calls
They must always work without authentication (CORS preflight requests)

### PR DESCRIPTION
**Pull request by kat-hi**

### CODE REVIEW

File: kubeportal/tests/test_api.py [hunk 1]
```diff
@@ -51,6 +51,13 @@ def post(self, relative_url, data=None, headers={}):
         return self.client.post('http://testserver' + relative_url,
                                 json=data, headers=headers)
 
+    def options(self, relative_url, headers={}):
+        if self.jwt:
+            headers["Authorization"] = None
+        if self.csrf:
```
Comment (troeger) [inline kubeportal/tests/test_api.py:57]: Is this the same as not setting them at all? If yes, why you do it? Please add an according comment in the source code.

File: kubeportal/tests/test_api.py [hunk 2]
```diff
@@ -160,6 +167,29 @@ def test_logout(self):
...
+
+        test_path = ['/api/',
+                     f'/api/{API_VERSION}/users/{self.admin.pk}',
+                     f'/api/{API_VERSION}/users/{self.admin.pk}/webapps',
+                     f'/api/{API_VERSION}/groups/{self.admin_group.pk}',
+                     f'/api/{API_VERSION}/cluster/k8s_apiserver',
+                     f'/api/{API_VERSION}/login/'
+        ]
+        for path in test_path:
+            if 'login' in path:
```
Comment (troeger) [inline kubeportal/tests/test_api.py:182]: Woah, ugly. What about putting the request method in the list above?

**Review by troeger — CHANGES_REQUESTED** [review]
Please check the inline comment. We need to get these header tests totally robust and complete, for less pain in the future.

File: kubeportal/tests/test_api.py (hunk 2 above)
Comment (kat-hi) [reply kubeportal/tests/test_api.py:182]: yeah it is the first time you describe my code as ugly. Great. thanks. :)

File: kubeportal/tests/test_api.py (hunk 1 above)
Comment (kat-hi) [reply kubeportal/tests/test_api.py]: well, actually yes. I just remembered funny networktools observations which led me to think that the key is necessary. In any case, it doesn't matter how the options-request looks like.

**Pull request merged**